js_modules/dagster-ui/packages/ui-core/src/asset-graph/sidebar/Sidebar.tsx

```diff
@@ -261,69 +373,98 @@ export const AssetGraphExplorerSidebar = React.memo(
-    indexOfLastSelectedNodeRef.current = indexOfLastSelectedNode;
 
-    React.useLayoutEffect(() => {
-      if (indexOfLastSelectedNode !== -1) {
-        rowVirtualizer.scrollToIndex(indexOfLastSelectedNode, {
-          align: 'center',
-          behavior: 'smooth',
+          const nodesInPath = path.split(':');
+          let currentPath = nodesInPath[0];
+
+          if (!currentPath) {
+            return prevOpenNodes;
+          }
+
+          const nextOpenNodes = new Set(prevOpenNodes);
+          nextOpenNodes.add(currentPath);
+          for (let i = 1; i < nodesInPath.length; i++) {
+            currentPath = `${currentPath}:${nodesInPath[i]}`;
+            nextOpenNodes.add(currentPath);
+          }
+          if (selectedNodeLeafToRoot?.id !== lastSelectedNode.id) {
+            setSelectedNodeLeafToRoot({id: lastSelectedNode.id, path: currentPath});
+          }
+          return nextOpenNodes;
         });
+      } else {
+        setSelectedNodeRootToLeaf(null);
+        setSelectedNodeLeafToRoot(null);
       }
-      // Only scroll if the rootNodes changes or the selected node changes
-      // otherwise opening/closing nodes will cause us to scroll again because the index changes
-      // if we toggle a node above the selected node
       // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [selectedNode, rootNodes, rowVirtualizer]);
+    }, [lastSelectedNode, graphData]);
+
+    const [expandedPanel, setExpandedPanel] = React.useState<'bottom' | 'top' | null>(null);
+
+    const isTopPanelHidden = expandedPanel === 'bottom';
+    const isBottomPanelHidden = expandedPanel === 'top';
+    const treeViewRows = useMemo(() => {
+      if (expandedPanel === 'top') {
+        return 'minmax(0, 1fr) auto';
+      } else if (expandedPanel === 'bottom') {
+        return 'auto minmax(0, 1fr)';
+      }
+      return 'repeat(2, minmax(0, 1fr))';
+    }, [expandedPanel]);
+
+    console.log({isTopPanelHidden, isBottomPanelHidden});
 
     return (
-      <div style={{display: 'grid', gridTemplateRows: 'auto minmax(0, 1fr)', height: '100%'}}>
+      <div
+        style={{
+          display: 'grid',
+          gridTemplateRows: 'auto auto minmax(0, 1fr)',
+          height: '100%',
+        }}
+      >
         <Box
           style={{
             display: 'grid',
             gridTemplateColumns: '1fr auto',
             gap: '6px',
-            padding: '12px 24px',
-            paddingRight: 12,
+            padding: '12px',
           }}
         >
+          <ButtonGroupWrapper>
+            <ButtonGroup
+              activeItems={new Set([sidebarViewType])}
+              buttons={[
+                {id: 'group', label: 'Group view', icon: 'asset_group'},
+                {id: 'tree', label: 'Tree view', icon: 'gantt_flat'},
+              ]}
+              onClick={(id: 'tree' | 'group') => {
+                setSidebarViewType(id);
+              }}
+            />
+          </ButtonGroupWrapper>
```

Comment:
I think this might look better with whitespace to it's right - right now when we stretch these button groups each item's text is left justified and it feels a bit odd? I had this same issue with one of the Billing UIs though, so maybe it's ok :-)